src/types/Config.ts

```diff
@@ -1,5 +1,17 @@
-export type Config = {
+type Config = {
   'module-markers'?: string[];
   'use-gitignore': boolean;
-  rules?: Record<string, boolean | Record<string, any>>;
+  rules?: Record<string, boolean | string | ConfigModuleRequiredFiles | ConfigPackageJsonSameName>;
 };
+
+type ConfigModuleRequiredFiles = {
```

Comment:
Nice!
We have to keep in mind that the number of typed here for the different rules will grow a lot

Reply:
Yep, I was thinking about that. Probably we will have to think in a way to group things, like "module related types" on a file and so on...